mentat/code_context.py

```diff
@@ -127,84 +97,70 @@ def set_code_map(self):
+        ctx = SESSION_CONTEXT.get()
 
-        stream.send("Code Context:", color="blue")
+        ctx.stream.send("Code Context:", color="blue")
         prefix = "  "
-        stream.send(f"{prefix}Directory: {git_root}")
-        if self.diff_context.name:
-            stream.send(f"{prefix}Diff:", end=" ")
-            stream.send(self.diff_context.get_display_context(), color="green")
+        ctx.stream.send(f"{prefix}Directory: {ctx.cwd}")
+        if self.diff_context and self.diff_context.name:
+            ctx.stream.send(f"{prefix}Diff:", end=" ")
+            ctx.stream.send(self.diff_context.get_display_context(), color="green")
         if self.include_files:
-            stream.send(f"{prefix}Included files:")
-            stream.send(f"{prefix + prefix}{git_root.name}")
+            ctx.stream.send(f"{prefix}Included files:")
+            ctx.stream.send(f"{prefix + prefix}{ctx.cwd.name}")
             print_path_tree(
-                build_path_tree(list(self.include_files.values()), git_root),
+                build_path_tree(list(self.include_files.values()), ctx.cwd),
                 get_paths_with_git_diffs(),
-                git_root,
+                ctx.cwd,
                 prefix + prefix,
             )
         else:
-            stream.send(f"{prefix}Included files: None", color="yellow")
-        auto = config.auto_tokens
+            ctx.stream.send(f"{prefix}Included files: None", color="yellow")
+        auto = ctx.config.auto_tokens
         if auto != 0:
-            stream.send(
-                f"{prefix}Auto-token limit:"
-                f" {'Model max (default)' if auto is None else auto}"
-            )
-            stream.send(
-                f"{prefix}CodeMaps: {'Enabled' if self.code_map else 'Disabled'}"
-            )
+            ctx.stream.send(f"{prefix}Auto-token limit:" f" {'Model max (default)' if auto is None else auto}")
+            ctx.stream.send(f"{prefix}CodeMaps: {'Enabled' if self.code_map else 'Disabled'}")
 
     def display_features(self):
         """Display a summary of all active features"""
-        session_context = SESSION_CONTEXT.get()
-        stream = session_context.stream
+        ctx = SESSION_CONTEXT.get()
 
         auto_features = {level: 0 for level in CodeMessageLevel}
         for f in self.features:
             if f.path not in self.include_files:
                 auto_features[f.level] += 1
         if any(auto_features.values()):
-            stream.send("Auto-Selected Features:", color="blue")
+            ctx.stream.send("Auto-Selected Features:", color="blue")
             for level, count in auto_features.items():
                 if count:
-                    stream.send(f"  {count} {level.description}")
+                    ctx.stream.send(f"  {count} {level.description}")
 
     _code_message: str | None = None
     _code_message_checksum: str | None = None
 
     def _get_code_message_checksum(self, max_tokens: Optional[int] = None) -> str:
```

Comment:
I just noticed this doesn't depend on the user prompt. Now that we're using embeddings and LLMs it should. @granawkins